### PR TITLE
Remove the one_personal_statement feature flag

### PR DESCRIPTION
**Pull request by JR-G**
## Context

At the end of April 2023 we merged subject knowledge and becoming a teacher into a single personal statement field (just using `becoming_a_teacher` on an `ApplicationForm`). During the 2023 recruitment cycle we were merging these fields on behalf of candidates that applied again, and from the start of 2024 we were merging them for any candidates who carried over.

We're now a couple of months into the 2024 recruitment cycle and as part of our wider tech debt clean up, it would be helpful to remove this now redundant code, and reduce further complexity.

## Changes proposed in this pull request

Removing subject knowledge and the `one_personal_statement` feature flag. The PR to remove this from analytics will follow. 

Arguably some of the personal statement fields can/should be moved in locales, and there's room for further renaming of becoming a teacher -> personal statement. I want to ideally get feedback first on how well the functionality holds up with the amount of files that are changed.

The vendor API functionality must be prioritised over everything else – removal of `subject_knowledge` **must not** cause a breaking change to any of the versions.

## Guidance to review

For the PR go commit-by-commit and for the functionality...

To check:
Candidate interface/Provider interface/Support interface:
(_can sign in, navigate to personal statement section, can edit, can view, can submit, can review, can be viewed by provider/support user, support can edit, providers can reject_)
- Fresh 2024 application works ok 
- Carried over pre-personal statement merger 2023 application works ok
- Carried over post-personal statement merger 2023 application works ok
- Pre-personal statement merger recruited 2023 application works ok

Vendor API:
- v1.0 calls can successfully retrieve applications in the above states
- v1.2 calls can successfully reject applications (without subject knowledge) in the above states

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/forms/candidate_interface/becoming_a_teacher_form.rb [hunk 1]
```diff
@@ -44,10 +39,8 @@ def save(application_form)
     end
 
     def presence_of_statement
```
Comment (avinhurry) [inline app/forms/candidate_interface/becoming_a_teacher_form.rb:41]: Do we still need this method?

File: spec/support/test_helpers/candidate_helper.rb [hunk 2]
```diff
@@ -79,11 +78,8 @@ def candidate_completes_application_form(with_referees: true, international: fal
     click_link(international ? 'Other qualifications' : 'A levels and other qualifications')
     candidate_fills_in_their_other_qualifications
 
-    click_link 'Why you want to teach'
-    candidate_fills_in_becoming_a_teacher
-
-    click_link 'Your suitability to teach a subject or age group'
-    candidate_fills_in_subject_knowledge
+    click_link 'Your personal statement'
+    candidate_fills_in_personal_statement
```
Comment (avinhurry) [inline spec/support/test_helpers/candidate_helper.rb:82]: Is the plan to rename `becoming_a_teacher` to `personal_statement` now?

**Review by avinhurry — APPROVED** [review]
Looks good, nice clean up! Couple of questions in line.

Will there be a follow up to remove the feature flag from the database?

File: app/forms/candidate_interface/becoming_a_teacher_form.rb (hunk 1 above)
Comment (JR-G) [reply app/forms/candidate_interface/becoming_a_teacher_form.rb:41]: Good shout raising this – I glossed over it but it turns out we're not validating the presence of the statement at all. I just tested locally on main and you can submit an app without a personal statement 🤔

**Review by vassyz — APPROVED** [review]
Looks good, amazing work!

**Pull request merged**